feat: support retry in `d/vsphere_network` (#2349)
hashicorp/terraform-provider-vsphere: website/docs/d/network.html.markdown

  the distributed virtual switch ID.
* `filter` - (Optional) Apply a filter for the discovered network.
  * `network_type`: This is required if you have multiple port groups with the same name. This will be one of `DistributedVirtualPortgroup` for distributed port groups, `Network` for standard (host-based) port groups, or `OpaqueNetwork` for networks managed externally, such as those managed by NSX.

> * `retry_timeout` - (Optional) If specified, the data source will retry read until timeout has elapsed. Timeout is specified in seconds.

Reviewer: ```suggestion
* `retry_timeout` - (Optional) The timeout duration in seconds for the data source to retry read operations.
```

Minor nit.

> * `retry_interval` - (Optional) Interval to retry the read in case `retry_timeout` is set. Specified in milliseconds, default is 500.

Reviewer: ```suggestion
* `retry_interval` - (Optional) The interval in milliseconds to retry the read operation if `retry_timeout` is set. Default: 500.
```

Minor nit.
